source/Missions/Services/Agents/Handlers/AgentDamageHandler.cs

```diff
@@ -0,0 +1,117 @@
+﻿using Common;
+using Common.Messaging;
+using Common.Network;
+using GameInterface.Serialization;
+using LiteNetLib;
+using Missions.Services.Agents.Messages;
+using Missions.Services.Agents.Packets;
+using Missions.Services.Network;
+using System;
+using TaleWorlds.MountAndBlade;
+
+namespace Missions.Services.Agents.Handlers
+{
+    /// <summary>
+    /// Handler for agent damage in a battle
+    /// </summary>
+    public interface IAgentDamageHandler : IHandler, IDisposable
+    {
+
+    }
+    /// <inheritdoc/>
+    public class AgentDamageHandler : IAgentDamageHandler
+    {
+        readonly INetworkAgentRegistry networkAgentRegistry;
+        readonly INetworkMessageBroker networkMessageBroker;
+        public AgentDamageHandler(INetworkAgentRegistry networkAgentRegistry, INetworkMessageBroker networkMessageBroker) 
+        {
+            this.networkAgentRegistry = networkAgentRegistry;
+            this.networkMessageBroker = networkMessageBroker;
+
+            networkMessageBroker.Subscribe<AgentDamage>(AgentDamageSend);
+            networkMessageBroker.Subscribe<NetworkAgentDamaged>(AgentDamageRecieve);
+        }
+        ~AgentDamageHandler()
+        {
+            Dispose();
+        }
+        public void Dispose()
+        {
+            networkMessageBroker.Unsubscribe<AgentDamage>(AgentDamageSend);
+            networkMessageBroker.Unsubscribe<NetworkAgentDamaged>(AgentDamageRecieve);
+        }
+
+        private void AgentDamageSend(MessagePayload<AgentDamage> payload)
+        {
+
+            // first, check if the attacker exists in the agent to ID groud, if not, no networking is needed (not a network agent)
+            if (networkAgentRegistry.TryGetAgentId(payload.What.AttackerAgent, out Guid attackerId) == false) return;
+
+            // next, check if the attacker is one of ours, if not, no networking is needed (not our agent dealing damage)
+            if (networkAgentRegistry.IsControlled(attackerId) == false) return;
+
+            networkAgentRegistry.TryGetAgentId(payload.What.VictimAgent, out Guid victimId);
```

Comment:
Add logging here when failing and return
```suggestion
            if (networkAgentRegistry.TryGetAgentId(payload.What.VictimAgent, out Guid victimId) == false)
            {
                Logger.Warning("Unable to get id for {agent} in {class}", payload.What.VictimAgent, typeof(AgentDamageHandler);
                return;
            }
```